engine/polyglot-api/src/main/java/org/enso/polyglot/debugger/IdExecutionService.java

```diff
@@ -69,6 +75,10 @@ public interface Callbacks {
      *     the execution and return the value as a result.
      */
     Object onFunctionReturn(Info info);
+
+    void setExecutionEnvironment(Info info);
```

Comment:
I see following code:
```
        Object result = callbacks.findCachedResult(info);

        if (result != null) {
          throw context.createUnwind(result);
        }
        callbacks.setExecutionEnvironment(info);
```
There is little need for this new callback method. Just do whatever is needed in `findCachedResult`.